server/base/src/test/java/org/apache/accumulo/server/conf/codec/VersionedPropEncryptCodecTest.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.server.conf.codec;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.security.NoSuchAlgorithmException;
+import java.time.Instant;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.crypto.Cipher;
+import javax.crypto.CipherInputStream;
+import javax.crypto.CipherOutputStream;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class VersionedPropEncryptCodecTest {
+
+  private final Logger log = LoggerFactory.getLogger(VersionedPropEncryptCodecTest.class);
+
+  /**
+   * Perform a round trip - encode, decode set of operations.
+   *
+   * @throws Exception
+   *           an exception is a test failure.
+   */
+  @Test
+  public void roundTripSample() throws Exception {
+
+    // set-up sample "secret" key - for testing only.
+    final char[] pass = {'a', 'b', 'c'};
+    final byte[] salt = {1, 2, 3};
+
+    VersionedPropEncryptCodec.GCMCipherParams cipherProps =
+        new VersionedPropEncryptCodec.GCMCipherParams(pass, salt);
+
+    Cipher cipher = Cipher.getInstance("AES/GCM/NoPadding");
```

Comment:
Using var to clean up redundant type specifications in local variables:
```suggestion
    var cipherProps = new VersionedPropEncryptCodec.GCMCipherParams(pass, salt);

    var cipher = Cipher.getInstance("AES/GCM/NoPadding");
```